src/urllib3/multipart/decoder.py

```diff
@@ -0,0 +1,165 @@
+"""Logic for parsing and decomposing a multipart response body."""
+import email.parser
+import typing
+
+from .. import _collections
+from .encoder import encode_with
+
+if typing.TYPE_CHECKING:
```

Comment:
I to make the typing compatible with other files, would suggest to don't use `typing.x` and importing them as `from typing import x,y,...`

Reply:
I find that importing things from the typing module pollutes namespaces. I can consolidate to that here but I really hate the practice of importing from typing rather than importing typing or aliasing the import typing to something shorter

Reply:
I can see both ways of (`import typing` or `from typing import x`) in different projects.
I am not against any of them. I would suggest going for one of them in all projects.

BTW, I am ok with `import typing` as well. I can update the rest of the files of the project in a PR.

@sethmlarson what do you think?

Reply:
Let's switch to the namedspaced usage with `typing.X`.

Reply:
We can use `libcst` to do it automagically for us if we want.